sql/parser/expr.go

```diff
@@ -402,6 +402,13 @@ type UnaryExpr struct {
 }
 
 func (node *UnaryExpr) String() string {
+	if node.Operator == UnaryMinus {
+		// We don't want to end up printing --5 since `--` indicates a comment.
+		// Instead, we drop both.
+		if nNode, ok := node.Expr.(*UnaryExpr); ok && nNode.Operator == UnaryMinus {
+			return nNode.Expr.String()
+		}
+	}
 	return fmt.Sprintf("%s%s", node.Operator, node.Expr)
```

Comment:
I'm not sure what is best, but I'd rather fix this with a space because `- -'foo'` should not become `'foo'`.